### PR TITLE
Allow using MRML via MRML-ruby

### PR DESCRIPTION
**Pull request by pokonski**
This proof of concept change would allow using Rust implementation of MJML called MRML via the mrml-ruby gem. 

The benefits:

- no need for JS runtime in production to compile email templates
- MRML boasts better performance and lower memory usage
- MRML-ruby gem comes with precompiled binaries for all architectures so much easier to set up
- no need to write to temp files to feed input into MJML

Would anyone except me find this useful?

References:

 - ruby gem: https://github.com/hardpixel/mrml-ruby
 - MRML: https://github.com/jdrouet/mrml

MRML[ isn't 100% compatible](https://github.com/jdrouet/mrml#missing-implementations) with MJML, so I made it optional.

### CODE REVIEW

**Comment by sighmon** [comment]
@pokonski The speed/lower memory use does look amazing. What's your feeling about the compromise of including `mlml-ruby` binaries when the default is not to use them? Should this be a fork?

If not, it'd be great if you could update the README and write a test when using the `use_mrml` flag.

I've also fixed the GitHub Action now for Ruby 2.7, so if you merge from `master` it should all work again.

**Comment by pokonski** [comment]
> What's your feeling about the compromise of including mrml-ruby binaries when the default is not to use them? Should this be a fork?

Good question, I am not opposed but ultimately we can make it a "peer" dependency and check if it's present inside the parser with `if defined?(MRML)` and raise error in that case.

> If not, it'd be great if you could update the README and write a test when using the use_mrml flag.

Will do, I didn't focus on tests initially because I wasn't sure if such addition is desired :)

**Comment by manuelmeurer** [comment]
Would be great to see this implemented! 😄

**Comment by pokonski** [comment]
@sighmon I updated the documentation and made `mrml` a "peer" dependency. Will add some tests too :)

**Comment by pokonski** [comment]
I also added tests for MrmlParser, CI is now green in my fork:

https://github.com/pokonski/mjml-rails/actions/runs/7488219746/job/20382217087

let me know if anything else is missing :)

File: lib/mjml.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 require 'mjml/handler'
 require 'mjml/parser'
-
+require 'mjml/mrml_parser' if defined?(MRML)
```
Comment (sighmon) [inline lib/mjml.rb:8]: This doesn't work for me in my Rails project even after installing `gem 'mrml'` and setting `config.use_mrml = true` in my initialiser.

When I comment out `if defined?(MRML)` it renders using MRML fine (very quickly!). Am I missing something else?

Here's the error I see:

```bash
NameError - uninitialized constant Mjml::MrmlParser:
  app/views/issues/email.mjml:1

```

**Review by sighmon — CHANGES_REQUESTED** [review]
Thanks so much for the README updates and test. One problem when installing it in my Rails project locally - do you have a demo project I can pull?

File: lib/mjml.rb (hunk 1 above)
Comment (pokonski) [reply lib/mjml.rb:8]: Looks like the order of gems in the `Gemfile` is important. I had `gem "mrml"` before `gem "mjml-rails"`, but if `mjml-rails` is first it fails. Maybe I can get rid of `if defined?(MRML)` completely, people will simply see an error if they forget to add `mrml` to their Gemfile.

Comment (pokonski) [reply lib/mjml.rb]: Okay I pushed a fix, can you check now?

**Review by sighmon — COMMENTED** [review]
(forgot to approve the review!)

**Comment by pokonski** [comment]
Awesome, thank you!

**Pull request merged**